### PR TITLE
[None][feat] Add PaliGemma2 (Gemma2 backbone) support to _torch path

### PR DESCRIPTION
**Pull request by shichiachi3-cyber**
Adds native PyTorch-backend support for Gemma2 text models and PaliGemma2 vision-language models in TensorRT-LLM's _torch path.

Key changes:
- **modeling_gemma2.py**: Gemma2ForCausalLM with alternating global/local sliding-window attention, attn_logit_softcapping, and final_logit_softcapping. Uses standard Attention (no QK norm) with query_pre_attn_scalar scaling.
- **modeling_paligemma.py**: PaliGemmaForConditionalGeneration combining SiglipVisionModel + linear projector + Gemma2ForCausalLM. Supports PaliGemma2 (gemma2 text backbone).
- **FlashInfer backend**: Add logits_soft_cap support to PlanParams and propagate through forward_impl → plan → wrapper.plan calls.
- **Attention base**: Add logits_soft_cap parameter, forwarded to the attention backend on every call.
- **RopeParams.from_config**: Handle rope_type="default" (used by Gemma2Config) as an alias for RotaryScalingType.none.
- **gemma2_weight_mapper.py**: HF weight mapper applying +1 correction for Gemma2's 1P RMSNorm convention. Registered for Gemma2ForCausalLM and PaliGemmaForConditionalGeneration.
- **tests**: Unit tests for Gemma2 model construction, alternating attention windows, softcap storage, and weight loading.

## Summary by CodeRabbit

* **New Features**
  * Added Gemma2 and PaliGemma multimodal models (image+text) with Gemma2 causal-LM support
  * Introduced an optional logits soft-cap control for attention behavior

* **Bug Fixes**
  * Robust rope-scaling handling for default/None values

* **Tests**
  * New unit tests validating Gemma2 model, attention sliding behavior, soft-cap persistence, and weight-loading transforms

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [ ] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Comment by shichiachi3-cyber** [comment]
@coderabbitai review

File: tensorrt_llm/_torch/attention_backend/interface.py [hunk 1]
```diff
@@ -515,8 +515,11 @@ def from_config(config) -> "RopeParams":
             rope_params.alpha = rope_scaling.get("alpha", 1.0)
             rotary_scaling_type = rope_scaling.get(
                 "type", None) or rope_scaling.get("rope_type")
-            rope_params.scale_type = RotaryScalingType.from_string(
-                rotary_scaling_type)
+            if rotary_scaling_type in (None, "default"):
```
Comment (pengbowang-nv) [inline tensorrt_llm/_torch/attention_backend/interface.py]: Maybe we should move this into RotaryScalingType class? Where does this `default` came from, would it behave differently for different models?

Comment (shichiachi3-cyber) [reply tensorrt_llm/_torch/attention_backend/interface.py]: Good call — moved into the enum in 0678119.

Where "default" comes from: it's HF's conventional rope_type value for unscaled RoPE. ROPE_INIT_FUNCTIONS["default"] is wired to _compute_default_rope_parameters in transformers/modeling_rope_utils.py (links pinned to v4.57.3, the version TRT-LLM main currently uses), which computes the standard RoPE formula (inv_freq = 1.0 / (base  ...), attention_factor = 1.0 — no scaling).

HF's modeling code also uses it as the runtime fallback whenever no rope_scaling is set; e.g., Gemma2RotaryEmbedding.__init__:

Python
if hasattr(config, "rope_scaling") and isinstance(config.rope_scaling, dict):
    self.rope_type = config.rope_scaling.get("rope_type", config.rope_scaling.get("type"))
else:
    self.rope_type = "default"
So "default" is HF's name for "plain RoPE." Anything that mirrors that convention into config.rope_scaling itself — the Gemma2 test fixture in this PR does exactly that — ends up with {"rope_type": "default"} reaching RotaryScalingType.from_string, and we want that accepted instead of raising.

Behavior across models: identical wherever it appears — _compute_default_rope_parameters is the standard RoPE for every model that uses it, so collapsing to RotaryScalingType.none is safe. The other named types in our enum (linear / dynamic / longrope / llama3 / yarn / mrope) keep the existing RotaryScalingType[s] lookup unchanged.

File: tensorrt_llm/_torch/modules/attention.py [hunk 2]
```diff
@@ -344,6 +344,7 @@ def __init__(
         use_custom_cublas_mm: bool = False,
         reduce_output: bool = True,
         mapping_with_cp: Optional[Mapping] = None,
+        logits_soft_cap: Optional[float] = None,
```
Comment (pengbowang-nv) [inline tensorrt_llm/_torch/modules/attention.py:348]: I think we should warn when backend is not flashinfer and logits_soft_cap is set, since default TRTLLM backend does not support this.

Comment (shichiachi3-cyber) [reply tensorrt_llm/_torch/modules/attention.py:348]: Good catch — addressed in 5f9fe19. Used `logger.warning_once` at `Attention.__init__` when `logits_soft_cap is not None and self.attn_backend != "FLASHINFER"`, mirroring the rope_fusion warning pattern at `attention.py:585-588`.

**Comment by shichiachi3-cyber** [comment]
Hi @pengbowang-nv — rebased onto main to clear the conflicts that came in with #12932 (Gemma4 multimodal) and #12829 (transformers 5.3.0).

  Notes:
  - `flashinfer.py`: kept upstream's `flashinfer_backend` parameter and `use_triton_prefill` fallback; threaded `logits_soft_cap` through both the new `if/else` branches in `forward_impl`.
  - `functional.py`: dropped my `from_string` change since upstream already handles `"default"` (with isinstance + lower() — strictly better than my version). The `RopeParams.from_config` simplification in `interface.py` is preserved.
  - Added a `[chore]` commit at the tip with pre-commit formatter output (isort + ruff-format), no behavior change.

Ready for your next pass whenever convenient.

**Comment by pengbowang-nv** [comment]
Attention part looks fine. If we want this to merge we would still need someone from modeling team to review this PR. Also  @laikhtewari could you please take a look at this PR because it adds a new model to our codebase, and evaluate if it is something we plan to support and maintain? Thanks!
